src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -1684,6 +1822,132 @@ UndistributeTable(Oid relationId)
 	{
 		ereport(ERROR, (errmsg("could not finish SPI connection")));
 	}
+
+	if (cascadeToColocated)
+	{
+		Oid colocatedTableId = InvalidOid;
+
+		/* For now we only support cascade to colocation for alter_distributed_table UDF */
+		Assert(conversionType == ALTER_DISTRIBUTED_TABLE);
+		foreach_oid(colocatedTableId, colocatedTableList)
+		{
+			if (colocatedTableId == relationId)
+			{
+				continue;
+			}
+			if (conversionType == ALTER_DISTRIBUTED_TABLE)
+			{
+				AlterDistributedTable(colocatedTableId, NULL, shardCount, relationName,
+									  false);
+			}
+		}
```

Comment:
in principle, we should be able to replay `GetReferencingForeignConstaintCommands` (should fix that typo) after this. 

ideally, we could separate foreign keys to reference tables (`GetForeignKeyOidsToReferenceTables`) from foreign keys to distributed tables. We can always recreate the former, and can do the latter when cascadeToColocated and not changing the distribution column.